libs/types/src/tokens.rs

```diff
@@ -255,6 +259,64 @@ impl CrossChainTransferability {
 	}
 }
 
+/// Connectors wrapped tokens
+///
+/// Currently, Connectors are only deployed on EVM-based chains and therefore
+/// the only supported type of connectors-wrapped tokens are Evm tokens. In the
+/// far future, we might support wrapped tokens from other chains, such as
+/// Cosmos, Avalanche, etc.
+#[derive(
+	Clone, Copy, PartialOrd, Ord, PartialEq, Eq, Debug, Encode, Decode, TypeInfo, MaxEncodedLen,
+)]
+#[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
+pub enum ConnectorsWrappedToken {
+	/// An EVM-native token
+	EVM {
+		/// The EVM chain id where the token is deployed
+		chain_id: EVMChainId,
+		/// The token contract address
+		address: [u8; 20],
+	},
+}
+
+impl TryFrom<MultiLocation> for ConnectorsWrappedToken {
+	type Error = ();
+
+	fn try_from(location: MultiLocation) -> Result<Self, Self::Error> {
+		match location {
+			MultiLocation {
+				parents: 0,
+				interior:
+					X2(
```

Comment:
Do we not need to support derivation for Moonbeam?

Reply:
How do you mean, @wischli? Moonbeam, from this perspective, would be seen as an EVM chain so that's already encapsulated in the `GlobalConsensus(NetworkId::Ethereum { chain_id }` part. See https://github.com/centrifuge/centrifuge-chain/pull/1393#discussion_r1224080128 for more context and info 👍

Reply:
Sorry, I missed out on that comment as it is not displayed in the `Files changed` section since the line it refers to is outdated.